flashrom.c

```diff
@@ -1449,7 +1449,7 @@ static int read_by_layout(struct flashctx *, uint8_t *);
 int read_flash_to_file(struct flashctx *flash, const char *filename)
 {
 	unsigned long size = flash->chip->total_size * 1024;
-	unsigned char *buf = calloc(size, sizeof(char));
+	unsigned char *buf = malloc(size);
```

Comment:
What's the warning?

This is killing the zero initialization.

Reply:
I missed the zero init, sorry. The warning is:

    ../../../flashrom.c:1452:23: warning: Result of 'calloc' is converted to a pointer of type 'unsigned char', which is incompatible with sizeof operand type 'char'
        unsigned char *buf = calloc(size, sizeof(char));
        ~~~~~~~~~~~~~~~      ^~~~~~       ~~~~~~~~~~~~

Maybe we just add an explicit cast?

Reply:
No need to mute the warning if you can fix it. I would try `sizeof(*buf)` or `sizeof(unsigned char)`.